### PR TITLE
Restore plain-text product search responses

### PR DESCRIPTION
**Pull request by kmanijak**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Product names returned by the AJAX search endpoints are currently HTML-escaped, causing characters such as ampersands and apostrophes to be returned as entities.

This PR restores plain-text product names by URL-decoding formatted names before stripping tags. This retains the released URL-decoding behavior while ensuring tags introduced during decoding are removed.

This restores the released response and `woocommerce_json_search_found_products` filter contract for plain-text product names.

Bug introduced in PR #68173.

### How to test the changes in this Pull Request:

1. Create products with the following names:
    - `Ben & Jerry's`
    - `Coffee%20Mug`
    - `Text <span>inside</span>`
    - `Text %3Cspan%3Einside%3C/span%3E`
2. Edit another product and search for each product under **Linked products → Upsells**.
3. Verify that the search results display:
    - `Ben & Jerry's`
    - `Coffee Mug`
    - `Text inside`
    - `Text inside`

### Testing that has already taken place:

- `pnpm test:php:env -- --filter WC_AJAX_Test::test_json_search_products_returns_plain_text_names`
- PHP syntax checks for both changed PHP files.
- Changed-file and full branch PHP coding-standard checks.
- PHPStan analysis of `includes/class-wc-ajax.php`.

### Milestone

- [x] Automatically assign milestone for the **[next WooCommerce version](../blob/trunk/plugins/woocommerce/woocommerce.php#L6)**

> **Note:** Check the box above to have the milestone automatically assigned when merged.
> Alternatively (e.g. for point releases), manually assign the appropriate milestone.

### Changelog entry

A patch changelog entry is included in this PR.

- [ ] Automatically create a changelog entry from the details below.
- [ ] This Pull Request does not require a changelog entry. (Comment required below)

### Release Communication

Select the release communication labels this PR needs:

- [ ] **Feature Highlight** - For user-facing features (what changed, user impact)
- [ ] **Developer Advisory** - For developer-facing changes (what changed, how to detect, actions needed)

### Use of AI Tools

Codex was used to investigate the regression, draft the code and tests, and prepare the pull request description. The resulting changes were reviewed and verified by the author.

### CODE REVIEW

File: plugins/woocommerce/includes/class-wc-ajax.php [hunk 1]
```diff
@@ -2137,7 +2137,7 @@ public static function json_search_products( $term = '', $include_variations = f
 				}
 			}//end if
 
-			$products[ $product_object->get_id() ] = esc_html( wp_strip_all_tags( $formatted_name ) );
+			$products[ $product_object->get_id() ] = wp_strip_all_tags( rawurldecode( $formatted_name ) );
```
[the remark this answers is not in the document]
Comment (kmanijak) [reply plugins/woocommerce/includes/class-wc-ajax.php:2140]: Thanks for flagging this. Decoding the entire formatted value is intentional here to preserve the behavior from before #68173: the endpoint applied `rawurldecode()` to the complete `get_formatted_name()` output, including the SKU. This PR only changes the ordering so decoding occurs before tag stripping. Restricting decoding to the product-name component would change the released output for percent-containing SKUs and would also require separately reconstructing and auditing formatted variation values. I would prefer to keep that behavior change outside this regression fix and consider it separately with dedicated compatibility coverage.

[the remark this answers is not in the document]
Comment (gigitux) [reply plugins/woocommerce/includes/class-wc-ajax.php:2140]: I think that it's a false positive because `rawurldecode` was present before https://github.com/woocommerce/woocommerce/pull/68173. @kmanijak WDYT?

**Review by gigitux — APPROVED** [review]
Thanks for digging into this! 🚀

File: plugins/woocommerce/includes/class-wc-ajax.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (kmanijak) [reply plugins/woocommerce/includes/class-wc-ajax.php:2140]: Yes, all good here 🙌
